src/GraphQL/Execution/DocumentExecutionListener.cs

```diff
@@ -1,77 +1,71 @@
 using System.Threading;
 using System.Threading.Tasks;
 using GraphQL.Validation;
+using GraphQL.Resolvers;
 
 namespace GraphQL.Execution
 {
+    /// <summary>
+    /// Provides the ability to log query validation failures and monitor progress of a GraphQL request's execution.
+    /// </summary>
     public interface IDocumentExecutionListener
     {
+        /// <summary>Executes after document validation is complete. Can be used to log validation failures.</summary>
         Task AfterValidationAsync(object userContext, IValidationResult validationResult, CancellationToken token);
+
+        /// <summary>Executes after document validation passes, before calling <see cref="IExecutionStrategy.ExecuteAsync(ExecutionContext)"/>.</summary>
         Task BeforeExecutionAsync(object userContext, CancellationToken token);
+
+        /// <summary>Executes before the <see cref="IDocumentExecuter"/> awaits the <see cref="Task"/> returned by <see cref="IExecutionStrategy.ExecuteAsync(ExecutionContext)"/></summary>
         Task BeforeExecutionAwaitedAsync(object userContext, CancellationToken token);
+
+        /// <summary>Executes after the <see cref="IExecutionStrategy"/> has completed executing the request</summary>
         Task AfterExecutionAsync(object userContext, CancellationToken token);
+
+        /// <summary>Executes before each time the <see cref="IExecutionStrategy"/> awaits the <see cref="Task{TResult}"/> returned by <see cref="IFieldResolver.Resolve"/></summary>
```

Comment:
Here we need to think about a more appropriate wording. `ParallelExecutionStrategy` can await multiple results. Besides `SubscriptionExecutionStrategy` does not call this method.

Reply:
I'll think about.  Note that SubscriptionExecutionStrategy inherits from ParallelExecutionStrategy, and the former calls the latter's ExecuteNodeTreeAsync method, which contains the call in question.  So it does get executed.

Reply:
```suggestion
        /// <summary>Executes before each time the <see cref="IExecutionStrategy"/> awaits the <see cref="Task{TResult}"/> returned by <see cref="IFieldResolver.Resolve"/>. For parallel resolvers, this may execute a single time prior to awaiting multiple tasks.</summary>
```
How about that?